### PR TITLE
Simplify internal function for mkdir -p.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Since Python 2 support has been removed, we can replace the internal function `_mkdir_p` with `os.makedirs(path, exist_ok=True)`.

## Motivation and Context
Simplifies internal code. Standard library features should be used where possible.

Cases where _files_ exist with the name for the destination directory raise an error. This was previously caught and tested separately but no longer needs special error handling code.

## Types of Changes
- [x] New feature

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

### CODE REVIEW

**Review by b-butler — APPROVED** [review]
Looks good :+1:

File: signac/contrib/utility.py [hunk 1]
```diff
@@ -274,19 +273,20 @@ def walkdepth(path, depth=0):
...
     path : str
         New directory name.
 
     """
-    try:
-        os.makedirs(path)
-    except OSError as error:
-        if not (error.errno == errno.EEXIST and os.path.isdir(path)):
-            raise
+    # Performance: `isdir` is fast and eliminates the need to check `_mkdir_p`
```
Comment (bdice) [inline signac/contrib/utility.py:284]:
```suggestion
    # Performance: `isdir` is fast and eliminates the rest of os.makedirs
```

**Comment by bdice** [comment]
@klywang Please review when you are able. Thanks!

**Review by klywang — APPROVED** [review]
Looks good!

**Pull request merged**